File: src/main/java/hylyht/adapter/servlet/resource/ServletResourceLoader.java
package hylyht.adapter.servlet.resource;

import com.google.common.base.Optional;

import javax.servlet.ServletContext;
import java.io.IOException;
import java.io.InputStream;

public class ServletResourceLoader implements ResourceLoader {

	private static final String DEFAULT_FILE_EXTENSION = "html";
	private final ServletContext servletContext;

	public ServletResourceLoader(ServletContext servletContext) {
		this.servletContext = servletContext;
	}

	@Override
	public Optional<? extends Resource> load(String pathInfo) throws IOException {
        MimeType mime = new ExtensionToMimeType().apply(parseFileExtension(pathInfo));
        InputStream in = servletContext.getResourceAsStream(pathInfo);
        if (in == null) return Optional.absent();
        return Optional.of(new StaticResource(in, mime));
	}

	private String parseFileExtension(String pathInfo) {
		String[] parts = pathInfo.split("\\.");
		if (parts.length > 1) return parts[parts.length - 1]; 
		return DEFAULT_FILE_EXTENSION;
	}

}

File: src/main/scala/hylyht/adapter/servlet/resource/ServletResourceLoader.scala
package hylyht.adapter.servlet.resource

import hylyht.adapter.servlet.ResourceLoader
import javax.servlet.ServletContext


object ServletResourceLoader {
  private val DEFAULT_FILE_EXTENSION = "html"
}

class ServletResourceLoader(
        private val servletContext: ServletContext) extends ResourceLoader
{

    def load(pathInfo: String): Option[Resource] = {
        val mime = new ExtensionToMimeType().apply(parseFileExtension(pathInfo))
        val in = servletContext.getResourceAsStream(pathInfo)
        if (in == null)
            None
        else
            Some(new StaticResource(in, mime))
    }

    private def parseFileExtension(pathInfo: String): String  = {
        val parts = pathInfo.split("\\.");
        if (parts.length > 1)
            parts(parts.length - 1)
        else
            ServletResourceLoader.DEFAULT_FILE_EXTENSION;
    }

}